UnitySamples/FBLeaderboard/Assets/FBLeaderboard/AccountSystem.cs

```diff
@@ -0,0 +1,290 @@
+	/// a general error or that the FB player has no ChilliConnect account. In the latter
+	/// case we link the current anonymous account to the current FB account.
+	/// 
+	/// @param error
+	/// 		Error type and description
+	/// 
+	private void OnChilliConnectFBLoginFailed(LogInUsingFacebookError error)
+	{
+		Debug.LogWarning("ChilliConnect logged in failed. Reason: " + error.ErrorDescription);
+
+		if(error.ErrorCode == LogInUsingFacebookError.Error.LoginNotFound)
+		{
+			LinkFacebookAccountRequestDesc requestDesc = new LinkFacebookAccountRequestDesc(m_fbAccessToken);
+			m_chilliConnect.PlayerAccounts.LinkFacebookAccount(requestDesc, (request, linkResponse) => OnChilliConnectLinked(linkResponse), (request, linkError) => OnChilliConnectLinkedFailed(linkError));
+		}
+	}
+
+	/// Called when a new ChilliConnect account is created. Once a new account is created you still have to login to it.
+	/// 
+	/// @param response
+	/// 	Holds the Id and Secret required for future anonymouse logins
+	///
+	private void OnChilliConnectAccountCreated(CreatePlayerResponse response)
+	{
+		Debug.Log("Created new CC account");
+
+		PlayerPrefs.SetString("CCId", response.ChilliConnectId);
+		PlayerPrefs.SetString("CCSecret", response.ChilliConnectSecret);
+
+		//Once we've created a new player we need to log them in
+		m_chilliConnect.PlayerAccounts.LogInUsingChilliConnect(response.ChilliConnectId, response.ChilliConnectSecret, (loginRequest) => OnChilliConnectAnonLoggedIn(), (loginRequest, error) => Debug.LogError(error.ErrorDescription));
+	}
+
+	/// Called when the request to link FB and Chilli accounts has completed successfully
+	///
+	/// @param response
+	/// 		Holds the response data including FB name
+	/// 
+	private void OnChilliConnectLinked(LinkFacebookAccountResponse response)
+	{
+		Debug.Log("ChilliConnect now linked with FB account for " + response.FacebookName);
+
+		m_localPlayerName = response.FacebookName;
+
+		//We consider logging in finished at this point.
+		OnAccountStatusChanged(AccountStatus.LOGIN_FB);
+	}
+
+	/// Called when the request to link FB and Chilli accounts has failed
+	/// There are generally cases where this will be called (connection issues, etc)
+	/// but an important case is if the current Chilli account is already linked to another FB account
+	/// 
+	/// @param error
+	/// 	Holds the error description
+	///
+	private void OnChilliConnectLinkedFailed(LinkFacebookAccountError error)
+	{
+		Debug.LogWarning("ChilliConnect link FB failed. Reason: " + error.ErrorDescription);
+
+		if(error.ErrorCode == LinkFacebookAccountError.Error.FacebookAccountLinkedWithAnotherPlayer)
+		{
+			//There are various solutions in this scenario:
+			// * You can replace the linked account with this one
+			// * You can logout of FB and prompt the user to login to the other account
+			// * You can switch to the ChilliAccount associated with FB (that is what this demo does).
```

Comment:
In fact I'm not sure this scenario can occur as we only attempt to link once the login to FB fails with "LoginNotFound" error. That means it wouldn't be possible to get FacebookAccountLinkedWithAnotherPlayer right?